Document the GitHub Packages classic-PAT requirement in README/CONTRIBUTING

Building Saiku from source requires a GitHub personal access token, and neither `README.md` nor `CONTRIBUTING.md` mentions it. Both currently have **zero** references to `read:packages`, `maven.pkg.github.com`, or `settings.xml`. It's documented only in `CLAUDE.md` and `docs/mondrian-fork.md` — neither of which a new contributor reads first.

The result: you follow the README, and the build dies with a bare 401 that never explains what's wrong.

```
[ERROR] Failed to execute goal on project saiku-olap-util: Could not collect dependencies
[ERROR] Failed to read artifact descriptor for pentaho:mondrian:jar:4.8.1.32
[ERROR]   Caused by: ... status code: 401, reason phrase: Unauthorized (401)
```

This has already cost us at least one self-hoster on Slack, and it's the kind of thing people hit silently and give up on.

### What the docs need to say

1. **Our Spicule artifacts live on GitHub Packages, which requires authentication for downloads even when the packages are public.** All 21 are `visibility=public`, so no org membership or `repo` scope is needed — but a token is still mandatory.

2. **It must be a _classic_ PAT with `read:packages`.** [GitHub only supports classic PATs for the Maven registry](https://docs.github.com/en/packages/working-with-a-github-packages-registry/working-with-the-apache-maven-registry). Fine-grained tokens 401 — and GitHub's UI now steers people to fine-grained by default, so this is the likely first failure.

3. **Five `<server>` entries are needed**, not four: `github-mondrian-saiku`, `github-olap4j`, `github-olap4j-xmlaserver`, `github-saiku-query`, `github-ossie`. The root pom registers five repos; `CLAUDE.md` documents four and omits `github-ossie` (see #1502). Simplest guidance: mirror the block CI writes in `.github/workflows/ci.yml`.

4. **A troubleshooting line**, because the failure mode is genuinely ambiguous: a fine-grained token and a classic token missing `read:packages` produce an *identical* bare 401 that never mentions scopes. Diagnose before re-minting:

   ```bash
   curl -sI -H "Authorization: token $PAT" https://api.github.com/user | grep -i x-oauth-scopes
   ```

   HTTP 200 with a scope list lacking `read:packages` means the token is valid but under-scoped. Classic PAT scopes are editable in place (Settings → Developer settings → Tokens (classic) → tick `read:packages` → Update token) — the value doesn't change, so `settings.xml` needs no edit.

### Suggested placement

A short "Building from source" section in `README.md` linking to fuller detail in `CONTRIBUTING.md`. Worth pairing with #1500 — together they're the whole reason a from-source build fails for anyone outside our setup.

--- a/README.md
+++ b/README.md
@@ -119,6 +119,24 @@ custom spans for `ThinQueryService` etc.).
 
 JDK 21 + Maven 3.9+ required.
 
+**Set up GitHub Packages auth first, or the build fails before it compiles.** Saiku's Mondrian fork, olap4j,
+saiku-query and Ossie artifacts are published to GitHub Packages, which requires an authenticated token **even
+though the packages are public**. Without it you get a bare `401 Unauthorized` on `pentaho:mondrian` that never
+mentions tokens:
+
+1. Create a **classic** personal access token with only the `read:packages` scope
+   ([Tokens (classic)](https://github.com/settings/tokens/new)). It must be classic — [GitHub's Maven registry
+   does not accept fine-grained tokens](https://docs.github.com/en/packages/working-with-a-github-packages-registry/working-with-the-apache-maven-registry),
+   and the UI defaults to fine-grained. Our packages are public, so no `repo` scope or org membership is needed.
+2. Add five `<server>` entries to `~/.m2/settings.xml` — `github-mondrian-saiku`, `github-olap4j`,
+   `github-olap4j-xmlaserver`, `github-saiku-query`, `github-ossie`. Copy the block from
+   [`.github/workflows/ci.yml`](.github/workflows/ci.yml); one token covers all five.
+
+Still getting a 401? A fine-grained token and a classic token missing `read:packages` produce an identical error,
+so check what the token actually has before minting another —
+`curl -sI -H "Authorization: token $PAT" https://api.github.com/user | grep -i x-oauth-scopes`. A classic token's
+scopes are editable in place, and the value doesn't change, so `settings.xml` needs no edit.
+
 ```sh
 # Compile, unit tests, Spotless format check (CI gate):
 mvn verify

